### ISSUE TITLE
Compiling using C

### ISSUE THREAD

**Issue by 1Mark** [joeyespo/sublimetext-console-exec#9]
Hi, 

I'm currently trying to get this to work with C (I use clang as my compiler) however I can't get it to work (I do use console exec for Java and that works fine.)

My sublime build for C is:

    {
    	"shell_cmd" : "clang-3.8.exe -o \"$file_base_name\" \"${file_name}\" && \"${file_base_name}.exe\"",
    	"selector" : "source.c",
    	"path" : "C:\\cygwin64\\bin;$path",
    	"working_dir" : "$file_path",
    	"target": "console_exec"
    }

Any ideas?

(I did try GCC just for the sake of trying to get it to work)
There is an improvement since it does open a terminal, but interestingly it closes immediately so nothing is seen.

    {
        "shell_cmd" : "gcc \"$file_name\" -o \"${file_base_name}.exe\" && \"${file_base_name}.exe\"",
        "selector" : "source.c",
        "path" : "C:\\cygwin64\\bin;$path",
        "working_dir" : "$file_path",
    	"target": "console_exec"
    }

**Comment by joeyespo** [comment]
Interesting.

A few questions. Any reason you're using `shell_cmd` over `cmd`? Is it because you have `&&`? I'm not sure that this will work because of how the plugin runs commands. (I can't remember the details off the top of my head. I'd have to dig deeper later.)

Could you share what you have for Java?

**Comment by 1Mark** [comment]
I used shell_cmd based on another build i found - no reason to be honest.

Ok I managed to figure it out - I'm not that familiar with JSON so I did google and find a different sublime-build and then modified it. The [original](https://www.quora.com/How-do-I-compile-and-execute-C-programs-not-C++-from-Sublime-Text-3)

And now my modification

    {
    	"cmd": ["gcc ${file_name} -o ${file_path}/${file_base_name} && ${file_path}/${file_base_name}.exe"],
    	"file_regex": "^(..[^:]*):([0-9]+):?([0-9]+)?:? (.*)$",
    	"working_dir": "${file_path}",
    	"selector": "source.c",
    	"target": "console_exec"
    }

So it works and all is solved. Thank you for the quick reply.
